tls/extensions/s2n_cert_status.c

```diff
@@ -80,7 +80,18 @@ int s2n_cert_status_recv(struct s2n_connection *conn, struct s2n_stuffer *in)
     uint8_t type;
     POSIX_GUARD(s2n_stuffer_read_uint8(in, &type));
     if (type != S2N_STATUS_REQUEST_OCSP) {
-        /* We only support OCSP */
+
+        /*
+         *= https://tools.ietf.org/rfc/rfc6066#section-8
+         *= type=exception
```

Comment:
This exception doesn't make sense to me. We are always validating the OCSP response. We would never not check it.

Reply:
spoke offline: explain that the exception is for `response is not satisfactory`

Reply:
Reason=For compatibility, we don't error if the response is not OCSP.

Reply:
I think you are right since we didnt receive a OCSP response.

> *# Clients requesting an OCSP response and receiving an OCSP response in
>  *# a “CertificateStatus” message MUST

However I am wondering since we did `request` a ocsp response (only reason we would process this message i believe), would this be considered a violation? Since we wont receive another CertificateStatus message and this one didnt have the satisfactory response we expected

Reply:
After some more offline discussion I dont believe this is an exception since the comment specifically talk about the `OCSP response`, which implies we got one and tried validating it.

It should be noted that we did negotiate an ocsp status, and the peer sent us a CertificateStatus message that did not contain a ocsp status. In this case the peer did misbehave. However, even then I dont think we would want to error for the sake of compatibility with wild implementations.